### PR TITLE
PUBDEV-7871 - Red Hat certification for H2O Docker image

### PR DESCRIPTION
**Pull request by Pscheidl**
https://h2oai.atlassian.net/browse/PUBDEV-7871
H2O Docker image has to be certified against Red Hat
- H2O Docker image is based on UBI (Red Hat Universal Base Image)
- OpenJDK 11 is installed. Tried detection of cgroup resources locally & manually just to be sure.
- Same image is used for operator. More lightweight. Will be pushed to hub.docker.com as well. This means starting this release, the official H2O open source docker image will be on UBI.
- Has to be pushed to Docker registry as well. For that, we need three secrets: `H2O_RED_HAT_REGISTRY_KEY`, `H2O_RED_HAT_PID`, and `H2O_RED_HAT_APIKEY`. Contacted @abal5 about adding them to Jenkins.
- Test releases are NOT pushed to RH at all

### CODE REVIEW

**Comment by Pscheidl** [comment]
Is there a way to test this @michalkurka ? E.g. putting the target temporarily into a TEST release and run it ? I can manually delete the artifact on RH side.

File: scripts/jenkins/red_hat/red_hat_docker_certification.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import argparse
+import sys
+import time
+
+import requests
+
+
+def has_scan_errors(digest, pid, api_key):
```
Comment (Pscheidl) [inline scripts/jenkins/red_hat/red_hat_docker_certification.py:8]: The script is Python 2.7 compatible, verified locally. Unsure which Python is in the release image.

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Dockerignore reducing time cost of context upload during docker build for docker/public/Dockerfile-h2o-release.
+# This Dockerfile requires whole H2O folder as context, as it uses multiple files across the project.
+
+# Ignore everything by default 
+*
```
Comment (honzasterba) [inline .dockerignore:5]: Personaly I am against this approach, could we just copy the LICENSE file to build dir via Makefile and and use the build dir still as the base for building the image?

File: scripts/jenkins/jenkinsfiles/Jenkinsfile-Release [hunk 3]
```diff
@@ -286,7 +286,10 @@ try {
                         try {
                             pipelineContext.getBuildSummary().addStageSummary(this, BUILD_H2O_DOCKER_STAGE_NAME, env.BUILD_NUMBER_DIR)
                             pipelineContext.getBuildSummary().setStageDetails(this, BUILD_H2O_DOCKER_STAGE_NAME, env.NODE_NAME, env.WORKSPACE)
-                            withCredentials([usernamePassword(credentialsId: 'dockerhub', passwordVariable: 'DOCKERHUB_PASSWORD', usernameVariable: 'DOCKERHUB_USERNAME')]) {
+                            withCredentials([usernamePassword(credentialsId: 'dockerhub', passwordVariable: 'DOCKERHUB_PASSWORD', usernameVariable: 'DOCKERHUB_USERNAME'),
```
Comment (honzasterba) [inline scripts/jenkins/jenkinsfiles/Jenkinsfile-Release:289]: I may be more readable to put the first `usernamePassword` at new line and the ending `]) { ` at newline as well

Comment (Pscheidl) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-Release]: Done.

File: .dockerignore (hunk 2 above)
Comment (Pscheidl) [reply .dockerignore]: Sure, we can. I don't really care about this part. Honoring the current way of doing things is also fine to me.

Done in. https://github.com/h2oai/h2o-3/pull/5197/commits/b7557168ed509d78e84d6492d2cfc3752d8cb2ce

**Review by honzasterba — APPROVED** [review]
lgtm

**Pull request merged**